core/src/sumeragi/message.rs

```diff
@@ -9,22 +9,9 @@ use crate::block::{CommittedBlock, ValidBlock};
 
 /// Helper structure, wrapping messages and view change proofs.
 #[derive(Debug, Clone, Decode, Encode)]
-pub struct MessagePacket {
-    /// Proof of view change. As part of this message handling, all
-    /// peers which agree with view change should sign it.
-    pub view_change_proofs: view_change::ProofChain,
-    /// Actual Sumeragi message in this packet.
-    pub message: Option<Message>,
-}
-
-impl MessagePacket {
-    /// Construct [`Self`]
-    pub fn new(view_change_proofs: view_change::ProofChain, message: Option<Message>) -> Self {
-        Self {
-            view_change_proofs,
-            message,
-        }
-    }
+pub enum MessagePacket {
+    ControlFlowMessage(ControlFlowMessage),
+    NormalMessage(Message),
```

Comment:
```suggestion
pub enum Message {
    ControlFlow(ControlFlowMessage),
    Block(BlockMessage),
```

- I think `Packet` suffix doesn't make sense anymore
- `Normal` prefix doesn't give anything to us
- No need to repeat `Message` in variants if enum name already contains it

Reply:
Good suggestion.